File: binary/python3/binary_tree_postorder_traversal.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def postorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        postorder = []
        
        if root is None:
            return postorder
        
        if root.left is not None:
            postorder += self.postorderTraversal(root.left)
            
        if root.right is not None:
            postorder += self.postorderTraversal(root.right)
            
        postorder.append(root.val)
        
        return postorder

File: binary/cpp/binary_tree_postorder_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        std::vector<int> answer;
        if (!root) {
            return answer;
        }
        if (root->left) {
            std::vector<int> left = postorderTraversal(root->left);
            answer.insert(answer.end(), left.begin(), left.end());
        }
        if (root->right) {
            std::vector<int> right = postorderTraversal(root->right);
            answer.insert(answer.end(), right.begin(), right.end());
        }
        answer.push_back(root->val);
        return answer;
    }
};
